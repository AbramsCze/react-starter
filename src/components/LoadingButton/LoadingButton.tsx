import React, { Component, Fragment } from 'react'
import { connect } from 'react-redux'
import { setCvLoading } from '../../actions/cv'

type StateToProps = {
  cvAreLoading?: boolean;
}

type DispatchToProps = {
  setCvLoading: (loading: boolean) => void;
}

export type Props = StateToProps & DispatchToProps

export class LoadingButton extends Component<Props> {
  constructor(props: Props) {
    super(props)
  }

  handleClick = (): void => {
    const { setCvLoading, cvAreLoading } = this.props
    setCvLoading(!cvAreLoading)
  }

  render() {
    const { cvAreLoading } = this.props
    const { handleClick } = this
    return (
      <Fragment>
        <button onClick={handleClick}>Toggle loading</button>
        <span>{cvAreLoading ? 'LOADING' : 'DONE'}</span>
      </Fragment>
    )
  }
}

const mapStateToProps = (state: State) => ({
  cvAreLoading: state.cv.loading
})

const mapDispatchToProps = {
  setCvLoading,
}

export default connect(mapStateToProps, mapDispatchToProps)(LoadingButton)

