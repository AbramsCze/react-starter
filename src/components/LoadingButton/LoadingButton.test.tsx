// libs
import React from 'react'
import { shallow } from 'enzyme'
import { fake } from 'sinon'

// others
import { LoadingButton, Props } from './LoadingButton'

describe('LoadingButton component', () => {
  let props: Props

  beforeEach(() => {
    props = {
      setCvLoading: fake(),
      cvAreLoading: false,
    }
  })

  it('should render', () => {
    const renderedComponent = shallow(<LoadingButton {...props} />)
    expect(renderedComponent).toMatchSnapshot()
  })
})
