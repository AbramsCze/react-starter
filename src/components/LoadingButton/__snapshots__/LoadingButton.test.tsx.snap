// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingButton component should render 1`] = `ShallowWrapper {}`;
